lib/python/ensembl/brc4/runnable/process_seq_region.py

```diff
@@ -190,34 +189,31 @@ def add_brc4_ebi_name(self, seq_regions: List[SeqRegion]) -> List[SeqRegion]:
             The seq_region_names data are directly added to the SeqRegion if found.
         """
         sources = self.param("brc4_synonym_sources")
-        sources_alt = self.param("brc4_synonym_sources_alt")
         
         new_seq_regions = []
         for seqr in seq_regions:
-            main_name = ''
-            alt_name = ''
+            names = dict()
             if "synonyms" in seqr:
                 for syn in seqr["synonyms"]:
-                    if syn["source"] in sources:
-                        main_name = syn["name"]
-                    if syn["source"] in sources_alt:
-                        alt_name = syn["name"]
+                    names[syn["source"]] = syn["name"]
             
-            name = ''
-            if main_name:
-                name = main_name
-            elif alt_name:
-                name = alt_name
-            else:
+            # Choose the synonym to use as the BRC name
+            # Use the first valid name from the source list
+            brc_name = ''
+            for source_name in sources:
+                if source_name in names:
+                    brc_name = names[source_name]
+                    break
+            
+            if not brc_name:
```

Comment:
This is an alternative implementation (Python's tricks!):
```suggestion
            for source_name in sources:
                if source_name in names:
                    brc_name = names[source_name]
                    break
            else:
```
Source: https://www.w3schools.com/python/gloss_python_for_else.asp